graylog2-web-interface/src/views/components/DashboardSearchBarForm.tsx

```diff
@@ -15,74 +15,66 @@
 import PropTypes from 'prop-types';
 import { Form, Formik } from 'formik';
-import { isFunction, isEmpty } from 'lodash';
+import { isFunction } from 'lodash';
 import type { FormikProps } from 'formik';
 
 import type { TimeRange } from 'views/logic/queries/Query';
-import validateTimeRange from 'views/components/TimeRangeValidation';
-import FormWarningsProvider from 'contexts/FormWarningsProvider';
+import FormWarningsContext from 'contexts/FormWarningsContext';
+import type { QueryValidationState } from 'views/components/searchbar/queryvalidation/types';
+import validate from 'views/components/searchbar/validate';
 
-import DateTimeProvider from './searchbar/date-time-picker/DateTimeProvider';
 import { onInitializingTimerange, onSubmittingTimerange } from './TimerangeForForm';
+import DateTimeProvider from './searchbar/date-time-picker/DateTimeProvider';
 
-type Values = {
+export type DashboardFormValues = {
   timerange: TimeRange | undefined | null,
   queryString: string | undefined | null,
 };
 
 type Props = {
-  initialValues: Values,
+  initialValues: DashboardFormValues,
   limitDuration: number,
-  onSubmit: (Values) => void | Promise<any>,
-  children: ((props: FormikProps<Values>) => React.ReactElement) | React.ReactElement,
+  onSubmit: (values: DashboardFormValues) => void | Promise<any>,
+  children: ((props: FormikProps<DashboardFormValues>) => React.ReactElement) | React.ReactElement,
+  validateQueryString: (values: DashboardFormValues) => Promise<QueryValidationState>,
 };
 
-const _isFunction = (children: Props['children']): children is (props: FormikProps<Values>) => React.ReactElement => isFunction(children);
-
-const validate = (nextTimeRange, limitDuration) => {
-  let errors = {};
-
-  const timeRangeErrors = validateTimeRange(nextTimeRange, limitDuration);
+const _isFunction = (children: Props['children']): children is (props: FormikProps<DashboardFormValues>) => React.ReactElement => isFunction(children);
 
-  if (!isEmpty(timeRangeErrors)) {
-    errors = { ...errors, timerange: timeRangeErrors };
-  }
-
-  return errors;
-};
-
-const DashboardSearchForm = ({ initialValues, limitDuration, onSubmit, children }: Props) => {
-  const _onSubmit = useCallback(({ timerange, queryString }) => {
+const DashboardSearchForm = ({ initialValues, limitDuration, onSubmit, validateQueryString, children }: Props) => {
+  const _onSubmit = useCallback(({ timerange, queryString }: DashboardFormValues) => {
     return onSubmit({
       timerange: Object.keys(timerange).length ? onSubmittingTimerange(timerange) : undefined,
       queryString,
     });
   }, [onSubmit]);
   const { timerange, queryString } = initialValues;
-  const initialTimeRange = timerange ? onInitializingTimerange(timerange) : {};
+  const initialTimeRange = timerange ? onInitializingTimerange(timerange) : {} as TimeRange;
   const _initialValues = {
     timerange: initialTimeRange,
     nextTimeRange: initialTimeRange,
```

Comment:
Not related to this PR, but I think we can remove `nextTimeRange` here.

Reply:
I wanted to take a look at this outside of the scope of this PR, so we can make sure that its effects are not related to the other refactorings.